xbmc/utils/URIUtils.cpp

```diff
@@ -44,6 +45,69 @@ using namespace XFILE;
 
 const CAdvancedSettings* URIUtils::m_advancedSettings = nullptr;
 
+namespace
+{
+
+constexpr std::string_view DecodeURLSpecialChars{"+%"};
+
+std::optional<char> DecodeOctlet(std::string_view& encoded)
+{
+  if (encoded.length() < 2)
+    return {};
+
+  uint8_t decimal{};
+  const auto res = std::from_chars(&encoded[0], &encoded[2], decimal, 16);
```

Comment:
If your length is only two the `[2]` de-references a value after the string. You have to use pointer arithmetic to create the end iterator.
```suggestion
  const auto res = std::from_chars(encoded.data(), encoded.data() + 2, decimal, 16);
```